### PR TITLE
fix: implemented is_shuffled and is_cards_distinct in DeckTrait

### LINKED ISSUE #178: [Feat][Doctor Needed] Implement `is_shuffled` and `is_cards_distinct` in `DeckTrait`
## Overview
There are two functions used when verifying the game. These two functions should be implemented so that more edge cases tests can be added to the tests in verifying game. These two functions should be implemented and tested. The target file for this task is `traits/deck.cairo`

The first function verifies if the deck is shuffled in the best way possible.
The second function verifies if all the cards in the deck are distinct.

May the force be with you.

## Help?
"I made it here and I will guide [this passage](https://github.com/Prometheus-A/The-Velvet-Ace/issues/178) all the way to the farthest ends of the earth. The stormiest seas will not stop me." _**--Pirate Gol D. Roger.**_

### PR DESCRIPTION
**Pull request by truthixify**
## Description   
Implemented is_
shuffled and is_cards_distinct in DeckTrait

## Related Issues   
Fixes #178 and #180 

## Changes Made   - [ ] 
- `is_shuffled` logic is implemented
- `is_cards_distinct` logic is implemented

## How to Test  
This can be tested by using them in gameplay

## Checklist  
- [x] My code follows the project's coding style.  
- [x] I have tested these changes locally.   
- [x] Documentation has been updated where necessary.

### CODE REVIEW

File: poker-texas-hold-em/contract/src/systems/actions.cairo [hunk 1]
```diff
@@ -245,8 +245,17 @@ pub mod actions {
                 game_pot += amount_to_call;
             }
 
+            let mut updated_game_pots: Array<u256> = ArrayTrait::new();
+            let mut i = 0;
+            while i != game_pots.len() - 1 {
+                updated_game_pots.append(*game_pots.at(i));
+                i += 1;
+            };
+            updated_game_pots.append(game_pot);
```
Comment (Birdmannn) [inline poker-texas-hold-em/contract/src/systems/actions.cairo:254]: Yoshh 💯

File: poker-texas-hold-em/contract/src/systems/actions.cairo [hunk 2]
```diff
@@ -620,8 +637,18 @@ pub mod actions {
...
+
+            let mut updated_game_pots: Array<u256> = ArrayTrait::new();
+            let mut i = 0;
+            while i != game_pots.len() - 1 {
+                updated_game_pots.append(*game_pots.at(i));
+                i += 1;
+            };
+            updated_game_pots.append(current_pot + amount);
+
+            world.write_member(Model::<Game>::ptr_from_keys(game_id), pot_, updated_game_pots);
```
Comment (Birdmannn) [inline poker-texas-hold-em/contract/src/systems/actions.cairo]: I think you'll have to revert this please, because `game_pots` already used `refresh_pots()`. The whole logic of `adjust_pot` is a whole lot, and it can be read in the issue description... In summary, it dynamically handles the values of the last two pots in the pots array when certain conditions are cleared. The adjustments of this last two pots has already occurred in the for loop prior to this.. No other adjustments should be made anymore. `self.refresh_pots()` does the justice of rearranging the whole pots involved in the right order, and returning a new array of pots which would be later written to the world.

File: poker-texas-hold-em/contract/src/traits/deck.cairo [hunk 3]
```diff
@@ -83,13 +83,65 @@ pub impl DeckImpl of DeckTrait {
...
+            while j != seen_cards.len() {
+                if *seen_cards.at(j) == key {
+                    // Cannot return here, so we break logic by setting flag
+                    distinct = false;
+                    break;
+                }
+                j += 1;
+            };
+
+            seen_cards.append(key);
```
Comment (Birdmannn) [inline poker-texas-hold-em/contract/src/traits/deck.cairo]: Wow, sorry about this, though mere looking at it, the loop process is extreme brute force. That said, you can use the logic in `count_unique_cards` in the tests in `models/deck.cairo`. You can shift that function into `utils/deck.cairo` and import the function into your function, and to the place you extracted it from.

Maybe I'll create a new issue on advanced shuffle check. Thank you very much

**Review by Birdmannn — COMMENTED** [review]
Holaa. Please have a look @truthixify

File: poker-texas-hold-em/contract/src/systems/actions.cairo (hunk 2 above)
Comment (Birdmannn) [reply poker-texas-hold-em/contract/src/systems/actions.cairo]: @truthixify I only said you should revert this, and it seemed you reverted all..

**Review by Birdmannn — APPROVED** [review]
Thank you very much @truthixify
